R/raincloudPlots.R

```diff
@@ -0,0 +1,908 @@
+  if (options$meanIntervalCustom) {
+    if (options$numberOfClouds == nrow(uniqueCombis)) {
+      output$lowerBound <- options$customizationTable[[2]]$values
+      output$upperBound <- options$customizationTable[[3]]$values
+      output$successfulComputation <- TRUE
+    } else {
+      # successfulComputation remains FALSE
+    }
+    return(output)
+  }
+
+  # Loop through each cell of the design...
+  for (cloudNumber in 1:nrow(uniqueCombis)) {
+    primaryLevel   <- as.character(uniqueCombis$primaryFactor[cloudNumber])    # as.character() ensures that it also works
+    secondaryLevel <- as.character(uniqueCombis$secondaryFactor[cloudNumber])  # for numeric factor levels
+    currentCell    <- dataset[dataset$primaryFactor == primaryLevel & dataset$secondaryFactor == secondaryLevel, ][[inputVariable]]
+
+    # ...and calculate lowerBound, upperBound accordingly
+    if (options$meanIntervalOption == "sd") {
+      xBar <- mean(currentCell)
+      sd   <- sd(currentCell)
+      output$sd[cloudNumber] <- sd
+      output$lowerBound[cloudNumber] <- xBar - sd
+      output$upperBound[cloudNumber] <- xBar + sd
+      output$successfulComputation <- TRUE
+
+    } else if (options$observationId == "" && options$meanIntervalOption == "ci" && options$meanCiAssumption) {
+      # The following is adapted from .descriptivesMeanCI() in jaspDescriptives
+
+      if (options$meanCiMethod == "normalModel") {
+        xBar <- mean(currentCell)
+        se <- sd(currentCell) / sqrt(length(currentCell))
+        z <- qnorm((1 - options$meanCiWidth) / 2, lower.tail = FALSE)
+        output$lowerBound[cloudNumber] <- xBar - z * se
+        output$upperBound[cloudNumber] <- xBar + z * se
+        output$successfulComputation <- TRUE
+
+      } else if (options$meanCiMethod == "oneSampleTTest") {
+        ttestResult <- stats::t.test(x = currentCell, conf.level = options$meanCiWidth)
+        CIs <- ttestResult[["conf.int"]]
+        output$lowerBound[cloudNumber] <- CIs[1]
+        output$upperBound[cloudNumber] <- CIs[2]
+        output$successfulComputation <- TRUE
+
+      } else if (options$meanCiMethod == "bootstrap") {
+        k <- options$meanCiBootstrapSamples
+        means <- numeric(k)
+        n <- length(currentCell)
+        jaspBase::.setSeedJASP(options)
+        for (i in seq_len(k)) {
+          bootData <- sample(currentCell, size = n, replace = TRUE)
+          means[i] <- mean(bootData)
+        }
+        percentiles <- (1 + c(-options$meanCiWidth, options$meanCiWidth)) / 2
+        CIs <- quantile(means, probs = percentiles)
+        output$lowerBound[cloudNumber] <- CIs[1]
+        output$upperBound[cloudNumber] <- CIs[2]
+        output$successfulComputation <- TRUE
+
+      } else {
+        # successfulComputation remains FALSE
+
+      }  # End section confidence interval
+
+    } else {
+      # successfulComputation remains FALSE
+
+    }  # End lowerBound, upperBound calculation
+  }  # End for loop through cells of the design
```

Comment:
I don't think the empty `else` statements are necessary, nor are the comments `} End xxx`.

Reply:
there are still empty else statements here, why?